src/units/abilities.cpp

```diff
@@ -218,6 +218,50 @@ unit_ability_list unit::get_abilities(const std::string& tag_name, const map_loc
 	return res;
 }
 
+unit_ability_list unit::get_abilities(const std::string& tag_name, const map_location& loc, const_attack_ptr weapon) const
```

Comment:
You've added a new function that's almost an exact copy of the function just above it. Instead, please alter this function so that it ignores the weapon if null, and delete the old function above it.